File: prototype/Assets/microcosmicWar/Scripts/item/ItemObjectImp.js

var itemObject:IitemObject;

function canUse(pGameObject:GameObject):boolean
{
	return itemObject.canUse(pGameObject);
}

virtual function use()
{
	itemObject.use();
}

function setItemObject(pItemObject:IitemObject)
{
	itemObject = pItemObject;
}

File: prototype/Assets/microcosmicWar/Scripts/item/ItemObjectImp.cs

using UnityEngine;
using System.Collections;

public class ItemObjectImp : MonoBehaviour
{

    public IitemObject itemObject;

    public bool canUse(GameObject pGameObject)
    {
        return itemObject.canUse(pGameObject);
    }

    public virtual void use()
    {
        itemObject.use();
    }

    public void setItemObject(IitemObject pItemObject)
    {
        itemObject = pItemObject;
    }
}